federation/apis/federation/types.go

```diff
@@ -108,3 +108,27 @@ type ClusterList struct {
 	// List of Cluster objects.
 	Items []Cluster `json:"items"`
 }
+
+// Temporary/alpha stuctures to support custom replica assignments within FederatedReplicaSet.
+
+// A set of preferences that can be added to federated version of ReplicaSet as a json-serialized annotation.
+// The preferences allow the user to express in which culsters he wants to put his replicas within the
+// mentiond FederatedReplicaSet.
+type FederatedReplicaSetPreferences struct {
+	// A maping between cluser names and preferences regarding local replicasets in these clusters.
+	// "*" applies to all clusters if explicint mapping is not provided. It is assumed that cluster
+	// should not have any replicas scheduled.
+	Clusters map[string]LocalReplicaSetPreferences `json:"clusters,omitempty"`
+}
+
+// Preferences regaring number of replicas assigned to a local replicaset within federated replica set.
+type LocalReplicaSetPreferences struct {
```

Comment:
nit: Local implies local to something. Given that the context here is the Federation, that might be confusing to others. Would ClusterReplicaSetPreferences be better?